src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -990,7 +990,10 @@ protected override void EndNestedResourceInfoWithContent(ODataNestedResourceInfo
             // resource set afterwards, we have to now close the array of links.
             if (navigationLinkScope.EntityReferenceLinkWritten && !navigationLinkScope.ResourceSetWritten && nestedResourceInfo.IsCollection.Value)
             {
-                this.jsonWriter.EndArrayScope();
+                if (!this.jsonLightOutputContext.WritingResponse)

```

Comment:
Why does this depend on whether or not we're writing a response. There are valid requests that take an array of entity references, enclosed in square brackets -- are the square brackets for that case written somewhere else?

Reply:
Yes. Writing the close square bracket in response is in `WriteEnd()` for resource set. We don't need to write `[` again here. 

Besides, `WriteEnd()` for resource set will process other part, for example, to write the nextpagelink annotation. 

---
In reply to: [285801842](https://github.com/OData/odata.net/pull/1464#discussion_r285801842) [](ancestors = 285801842)